Products/Zuul/catalog/model_catalog_tool.py

```diff
@@ -270,6 +271,17 @@ def getValue(obj):
         return sorted((unbrain(brain) for brain in queryResults),
                       key=getValue, reverse=reverse, cmp=natural_compare)
 
+    def cursor_search(self, types=(), limit=DEFAULT_SEARCH_LIMIT, filterPermissions=False, fields=None):
+        """
+        Get all results from catalog by using solr deep pagination.
+        return generator.
+        """
+        init_cursor = INITIAL_CURSOR_MARK
+        order_by = INDEX_UNIQUE_FIELD
+        query, _ = self._build_query(filterPermissions=filterPermissions, types=types)
+        search_params = SearchParams(query, limit=limit, order_by=order_by, next_cursor=init_cursor, fields=fields)
```

Comment:
I think we can further constrain (or simplify) the API by having modelindex provide a `CursorSearchParams` (having `SearchParams` as a base class) object that fills in the `order_by` and `next_cursor` fields with the correct values.  This will complete the symmetry of the search methods and strongly indicate that they are different.  The `search` method uses `SearchParams` and the `cursor_search` method uses `CursorSearchParams`.